### ISSUE TITLE
LightGBM imbalanced data predict_proba result only 1.0 and 0

### ISSUE THREAD

**Issue by Zach-sis** [microsoft/LightGBM#6229]
## Description

When I run the imbalanced data with is_unbalance=True,
I got the predict_proba result only 1.0 or 0.0 for positive label
As my knowledge this function will get the probability which sum=1 and I sure I don't put the feature label in my input
I wander to konw what happend about this and how to solve it

thanks

## Environment info

LightGBM version or commit hash:

Command(s) you used to install LightGBM

```shell

```

**Comment by jmoralez** [comment]
Hey @Zach-sis, thanks for using LightGBM. Can you please provide a reproducible example? The following works fine:
```python
import lightgbm as lgb
from sklearn.datasets import make_classification

X, y = make_classification(n_samples=1_000, weights=[0.999])
y.mean()  # 0.007
model = lgb.LGBMClassifier(n_estimators=5, num_leaves=16, is_unbalance=True).fit(X, y)
probas = model.predict_proba(X)
np.unique(probas[:, 1]).size  # 53
```

**Comment by Zach-sis** [comment]
Hello, this is my model code which my imbalance data is 99.8% : 0.2 %

![image001](https://github.com/microsoft/LightGBM/assets/71366787/f4d7d435-eefc-4c2e-bb16-ab68506d3967)

then the follow code is the predict_prob which I change to dataframe for reading
![image003 (1)](https://github.com/microsoft/LightGBM/assets/71366787/9989cb48-eaa1-4717-9f2a-fb967317c5ac)


this is my first 10 results then it only 1.0 and 0.0
![unnamed](https://github.com/microsoft/LightGBM/assets/71366787/e279bbe7-abca-4d3a-add0-cb729df6b658)

and if I down to the bottom of X_test, the probability of will change to 0.0 and 1.0

there is not any probability between 1.0 and 0.0 , and I wander to know why

Also, I tune some hyperparameter, the results will became same probability like this

![unnamed (1)](https://github.com/microsoft/LightGBM/assets/71366787/be59c79d-2857-4edf-a017-8c68a162d2db)

hope this problem can be down, thanks

**Comment by jmoralez** [comment]
I don't think this is a problem with LightGBM. Setting `is_unbalance=True` will generate sample weights according to the class proportions, so it'll give more weight to the imbalanced class. In your case it seems like it's producing shallow trees because it can't find any more splits that produce a positive gain (you're probably getting a bunch of warnings about it).

**Comment by Zach-sis** [comment]
Thanks for your reply, I will try on other algorithm and check my abt
